homeassistant/components/influxdb/sensor.py

```diff
@@ -70,34 +75,95 @@
 def _merge_connection_config_into_query(conf, query):
     """Merge connection details into each configured query."""
     for key in conf:
-        if key not in query and key not in [CONF_QUERIES, CONF_QUERIES_FLUX]:
+        if key not in query and key not in [
+            CONF_QUERIES,
+            CONF_QUERIES_FLUX,
+            CONF_QUERIES_RAW,
+        ]:
             query[key] = conf[key]
 
 
+def validate_raw_version_config(conf: dict) -> dict:
+    """Ensure correct config fields are provided for queries_raw based on API version used."""
+
+    def __is_valid(setting):
+        if conf[CONF_API_VERSION] == API_VERSION_2 and CONF_DB_NAME in setting:
+            _LOGGER.error(
+                QUERIES_RAW_BASE_ERROR,
+                setting.get(CONF_NAME),
+                CONF_DB_NAME,
+                CONF_API_VERSION,
+                API_VERSION_2,
+            )
+            return False
+
+        if conf[CONF_API_VERSION] == DEFAULT_API_VERSION and CONF_BUCKET in setting:
+            _LOGGER.error(
+                QUERIES_RAW_BASE_ERROR,
+                setting.get(CONF_NAME),
+                CONF_DB_NAME,
+                CONF_API_VERSION,
+                DEFAULT_API_VERSION,
+            )
+            return False
+
+        return True
+
+    if CONF_QUERIES_RAW in conf:
+        conf[CONF_QUERIES_RAW][:] = [
+            query for query in conf[CONF_QUERIES_RAW] if __is_valid(query)
+        ]
```

Comment:
You need to call `_merge_connection_config_into_query` on each item in the `queries_raw` array just like it does [here for queries_flux](https://github.com/home-assistant/core/blob/dev/homeassistant/components/influxdb/sensor.py#L89) and [here for queries](https://github.com/home-assistant/core/blob/dev/homeassistant/components/influxdb/sensor.py#L101).

Otherwise it will only work if you provide a value for `database` in each item in `queries_raw`. A config like this:
```yaml
api_version: 1
database: my_default_database
queries_raw:
  - name: Test query 
    query: Test query
```
Won't work right now. `Test query` should inherit the value provided for `database` at the top level since it doesn't override it but since you aren't merging those details in then it won't work.

Reply:
Also your code seems to only return invalid if every item in `queries_raw` is invalid. But if there are one or two valid ones and a bunch of invalid ones then it will proceed here. It should not validate a partially valid config, its either all valid or all not.

Reply:
> You need to call `_merge_connection_config_into_query` on each item in the `queries_raw` array just like it does [here for queries_flux](https://github.com/home-assistant/core/blob/dev/homeassistant/components/influxdb/sensor.py#L89) and [here for queries](https://github.com/home-assistant/core/blob/dev/homeassistant/components/influxdb/sensor.py#L101).

Yes, it normally does in `validate_query_format_for_version` as for `queries` or `queries_flux`, here for [queries_raw API V1](https://github.com/home-assistant/core/blob/dev/homeassistant/components/influxdb/sensor.py#L158) and here for [queries_raw API V2](https://github.com/home-assistant/core/blob/dev/homeassistant/components/influxdb/sensor.py#L137) 
          
I had in mind to remove `queries_raw` sensors with invalid configuration due to the configured API mode (for instance `database` in a `queries_raw` sensor whereas influxDB is configured in API V2 mode.
 the need to check there isn't any problem in config for a `queries_raw` sensor, like a `bucket` field whereas influxDB is configured in API V1 mode or `database` whereas influxDB is configured in API V2 mode. I found that more usefull than raising a `vol.Invalid` as this doesn't prevent other sensors with correct config to be setup.

By the way, as `field` and `bucket` options are removed and to be consistent with the general behavior in case of invalid configuration, this part of code will be redifined.